relayer/src/event/rpc.rs

```diff
@@ -142,48 +147,36 @@ pub fn get_all_events(
             .map_err(|_| String::from("tx_result.height: invalid header height of 0"))?;
 
             for abci_event in &tx_result.result.events {
-                if query == queries::ibc_client().to_string() {
-                    if let Some(mut client_event) = events::client::try_from_tx(abci_event) {
-                        client_event.set_height(height);
-                        tracing::trace!("extracted ibc_client event {}", client_event);
-                        vals.push((height, client_event));
-                    }
-                }
-                if query == queries::ibc_connection().to_string() {
-                    if let Some(mut conn_event) = events::connection::try_from_tx(abci_event) {
-                        conn_event.set_height(height);
-                        tracing::trace!("extracted ibc_connection event {}", conn_event);
-                        vals.push((height, conn_event));
-                    }
-                }
-                if query == queries::ibc_channel().to_string() {
-                    if let Some(mut chan_event) = events::channel::try_from_tx(abci_event) {
-                        chan_event.set_height(height);
+                if let Ok(ibc_event) = IbcEvent::try_from(abci_event) {
+                    if query == queries::ibc_channel().to_string() {
                         let _span = tracing::trace_span!("ibc_channel event").entered();
-                        tracing::trace!("extracted {}", chan_event);
-                        if matches!(chan_event, IbcEvent::SendPacket(_)) {
+                        tracing::trace!("extracted {}", ibc_event);
+                        if matches!(ibc_event, IbcEvent::SendPacket(_)) {
                             // Should be the same as the hash of tx_result.tx?
                             if let Some(hash) =
                                 events.get("tx.hash").and_then(|values| values.get(0))
                             {
                                 tracing::trace!(event = "SendPacket", "tx hash: {}", hash);
                             }
                         }
-                        vals.push((height, chan_event));
+                    } else {
+                        tracing::trace!("extracted ibc event {}", ibc_event);
                     }
+
+                    events_with_height.push(IbcEventWithHeight::new(ibc_event, height));
```

Comment:
We previously only returned events based on the query and now we're returning all `IbcEvent`s in the tx result. Could this have unintended effects elsewhere?

Reply:
Good catch! [a60e524](https://github.com/informalsystems/ibc-rs/pull/2542/commits/a60e52483df7d3e11f7238464ed81a425bac5d2d)

Reply:
Try 2: [377b60b](https://github.com/informalsystems/ibc-rs/pull/2542/commits/377b60bbf963d670eb969bb72cdc6392339f3d0f)

Reply:
https://github.com/informalsystems/ibc-rs/pull/2542/files#r943587248